Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -2681,4 +2686,58 @@ void WebProcessPool::updateWebProcessSuspensionDelayWithPacing(WeakHashSet<WebPr
 
 #endif // ENABLE(WEB_PROCESS_SUSPENSION_DELAY)
 
+#if ENABLE(CONTENT_EXTENSIONS)
+
+constexpr static Seconds resourceMonitorRuleListCheckInterval = 24_h;
+
+WebCompiledContentRuleList* WebProcessPool::cachedResourceMonitorRuleList()
+{
+    if (!m_resourceMonitorRuleListCache)
+        loadOrUpdateResourceMonitorRuleList();
+
+    return m_resourceMonitorRuleListCache.get();
+}
+
+void WebProcessPool::loadOrUpdateResourceMonitorRuleList()
+{
+    if (m_resourceMonitorRuleListLoading)
+        return;
+
+    WEBPROCESSPOOL_RELEASE_LOG(Process, "loadOrUpdateResourceMonitorRuleList: rule list is requested");
+
+    m_resourceMonitorRuleListLoading = true;
+
+    platformLoadResourceMonitorRuleList([weakThis = WeakPtr { *this }] {
+        RefPtr protectedThis = weakThis.get();
+        if (!protectedThis)
+            return;
+
+        protectedThis->m_resourceMonitorRuleListLoading = false;
+
+        RefPtr ruleList = protectedThis->m_resourceMonitorRuleListCache;
```

Comment:
On non-Darwin platforms, `platformLoadResourceMonitorRuleList` always returns nullptr. This means that every time `cachedResourceMonitorRuleList` runs, it does some unnecessary work to try to load an empty rule list.

It might be nice to optimize this out, e.g. by making sure that you only call platformLoadResourceMonitorRuleList or loadOrUpdateResourceMonitorRuleList at most once a day.